src/libsystemd/sd-journal/journal-send.c

```diff
@@ -58,7 +58,8 @@ static int journal_fd(void) {
 
         fd_inc_sndbuf(fd, SNDBUF_SIZE);
 
-        if (!__sync_bool_compare_and_swap(&fd_plus_one, 0, fd+1)) {
+        if (!__atomic_compare_exchange_n(&fd_plus_one, &(int){0}, fd+1,
```

Comment:
please make this `&(const int) {0}`, so that the compiler can share the value with other stuff.

Reply:
This won't work actually, because the second parameter must now be writable, hence why I couldn't just use a literal `0`.

If I add that I get:
```
../src/libsystemd/sd-journal/journal-send.c:62:17: warning: passing argument 2 of ‘__atomic_compare_exchange_4’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
```